src/briefcase/commands/dev.py

```diff
@@ -164,6 +189,11 @@ def __call__(
             write_dist_info(app, dist_info_path)
 
         if run_app:
-            self.logger.info("Starting in dev mode...", prefix=app.app_name)
-            env = self.get_environment(app)
-            return self.run_dev_app(app, env, **options)
+            if test_mode:
+                self.logger.info(
+                    "Running test suite in dev environment...", prefix=app.app_name
+                )
```

Comment:
The user experience for `briefcase dev --test` for an app that's improperly configured is a little rough.

Do you think some guardrails here and exception masking with more obvious feedback to the user is appropriate?

```
❯ briefcase dev --test

[testmode] Running test suite in dev environment...
Traceback (most recent call last):
  File "/home/russell/.pyenv/versions/3.10.8-debug/lib/python3.10/runpy.py", line 129, in _get_module_details
    spec = importlib.util.find_spec(mod_name)
  File "/home/russell/.pyenv/versions/3.10.8-debug/lib/python3.10/importlib/util.py", line 94, in find_spec
    parent = __import__(parent_name, fromlist=['__path__'])
ModuleNotFoundError: No module named 'tests'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/russell/.pyenv/versions/3.10.8-debug/lib/python3.10/runpy.py", line 220, in run_module
    mod_name, mod_spec, code = _get_module_details(mod_name)
  File "/home/russell/.pyenv/versions/3.10.8-debug/lib/python3.10/runpy.py", line 138, in _get_module_details
    raise error(msg.format(mod_name, type(ex).__name__, ex)) from ex
ImportError: Error while finding module specification for 'tests.testmode' (ModuleNotFoundError: No module named 'tests')
```

Reply:
Hrm - I agree this isn't a great UX. However, there are some limits on what we can do here, because there's functionally no difference between user-space code raising an `ImportError`, and the main execution script raising an `ImportError`. We *could* catch the ImportError, and then interrogate to see if the underlying cause is the module that was used to start the app; however, that kind of interrogation makes me really nervous - being "smart" in this kind of way has burned me in the past.

However - I noticed that I'm inadvertently eating the error condition; The mainline essentially ignores `BriefcaseTestSuiteError`, assuming the method that raised it has done all the error handling, so when your test suite raises an error, that's not caught as a "couldn't start test suite" error. Catching that error at least gives us the chance to raise a slightly more meaningful "Unable to start test suite 'tests.myapp'" message, and capture the log; test suite success/failure is a "successful" run. 

Does that look any better?